src/librustc_typeck/diagnostics.rs

```diff
@@ -526,9 +526,10 @@ impl Foo {
 
 fn main() {
     let foo = Foo;
-    let ref_foo = &&Foo;
+    let ref_foo = &&&&&&&&Foo;
 
-    // error, reached the recursion limit while auto-dereferencing `&&Foo`
+    // error, reached the recursion limit while
+    // auto-dereferencing `&&&&&&&&Foo`
```

Comment:
what's up with this?

Reply:
I think this was a necessary change when this was based on some older versions of uninhabitedness checking. I can revert it now.